src/modules/python/pbs_hooks/PBS_power.PY

```diff
@@ -287,6 +287,31 @@ if this_event.type == pbs.PERIODIC:
                         exec_vnodes[vn]["neededby"] = start_time
         i += 1
 
+    try:
+        pbs_conf = pbs.get_pbs_conf()
+        pbs_home = pbs_conf['PBS_HOME']
+    except Exception:
+        pbs.logmsg(pbs.EVENT_DEBUG,
+                   "PBS_HOME needs to be defined in the config file")
+        pbs.logmsg(pbs.EVENT_DEBUG, "Exiting the power hook")
+        pbs.event().accept()
+
+    # Identify the nodes file and read in the data
+    node_file = ''
+    sleep_node_list = []
+    node_file = os.path.join(pbs_home, 'server_priv', 'hooks', 'tmp',
+                              'pbs_power_nodes_file')
+    if os.path.isfile(node_file) and os.stat(node_file).st_size:
+        pbs.logmsg(pbs.EVENT_DEBUG3, "pbs_power_nodes_file is %s" % node_file)
+        try:
+            fd = open(node_file, 'r')
+            sleep_node_list = fd.read().split(',')
+            fd.close()
+        except IOError:
+            raise Exception("I/O error reading node file")
+        except:
+            raise Exception("Error reading node file")
```

Comment:
Don't use a blanket except.  Narrow down the scope.

Reply:
Done.